### PR TITLE
Improve logprobs check in LiteLLMModel

### PR DESCRIPTION
**Pull request by alexandonian**
This pull request includes a fix to the `acompletion_iter` method in `lmi/llms.py` to ensure proper handling of `logprobs` attributes when processing completion choices. Some combinations of `litellm` version and providers appear to optionally return a `ChoiceLogprobs` object, which needs to be handled.

### Bug Fix:
* [`packages/lmi/src/lmi/llms.py`](diffhunk://#diff-8d96b00faaed994af6947056766b491a67899adf87382d9e02d9d7a976bfa5f3L756-R760): Updated the condition to check for the existence and validity of the `logprobs` attribute and its `content` property before accessing it, preventing potential errors when these attributes are missing or `None`.

### CODE REVIEW

File: packages/lmi/src/lmi/llms.py [hunk 1]
```diff
@@ -753,7 +753,7 @@ async def acompletion_iter(
                 used_model = completion.model or self.name
             choice = completion.choices[0]
             delta = choice.delta
-            if hasattr(choice.logprobs, "content"):
+            if getattr(choice.logprobs, "content", None):
```
Comment (jamesbraza) [inline packages/lmi/src/lmi/llms.py]: Can you add a comment saying that sometimes `"content"` is not an attribute, and sometimes it's `NoneType`

**Review by jamesbraza — COMMENTED** [review]
Any chance you can add a unit test 🤓

File: packages/lmi/src/lmi/llms.py (hunk 1 above)
Comment (alexandonian) [reply packages/lmi/src/lmi/llms.py]: If you don't mind the walrus operator, an intermediate `logprob_content` variable helps to satisfy the type checker and perhaps is more readable.
```python
if logprob_content := getattr(choice.logprobs, "content", None):
    logprobs.append(logprob_content[0].logprob or 0)
```

File: packages/lmi/tests/test_llms.py [hunk 2]
```diff
@@ -391,6 +391,90 @@ def test_pickling(self, tmp_path: pathlib.Path) -> None:
...
+            usage: Any = None,
+        ) -> Mock:
+            mock_completion = Mock()
+            mock_completion.model = model
+            mock_completion.choices = [Mock()]
+            mock_completion.choices[0].logprobs = logprobs
+            mock_completion.choices[0].delta = Mock()
+            mock_completion.choices[0].delta.content = delta_content
+            mock_completion.choices[0].delta.reasoning_content = delta_reasoning_content
+            mock_completion.choices[0].delta.role = delta_role
```
Comment (jamesbraza) [inline packages/lmi/tests/test_llms.py]:
```suggestion
            mock_completion.choices[0].delta = Mock(
                content=delta_content,
                reasoning_content=delta_reasoning_content,
                role=delta_role,
            )
```

File: packages/lmi/tests/test_llms.py [hunk 3]
```diff
@@ -391,6 +391,90 @@ def test_pickling(self, tmp_path: pathlib.Path) -> None:
...
+            logprobs: Any = None,
+            delta_content: str = "",
+            delta_reasoning_content: str = "hmmm",
+            delta_role: str = "assistant",
+            usage: Any = None,
+        ) -> Mock:
+            mock_completion = Mock()
+            mock_completion.model = model
+            mock_completion.choices = [Mock()]
+            mock_completion.choices[0].logprobs = logprobs
```
Comment (jamesbraza) [inline packages/lmi/tests/test_llms.py]:
```suggestion
            mock_completion.choices = [Mock(logprobs=logprobs)]
```

File: packages/lmi/tests/test_llms.py [hunk 4]
```diff
@@ -391,6 +391,90 @@ def test_pickling(self, tmp_path: pathlib.Path) -> None:
         assert llm.config == rehydrated_llm.config
         assert llm.router.deployment_names == rehydrated_llm.router.deployment_names
 
+    @pytest.mark.asyncio
+    async def test_acompletion_iter_logprobs_edge_cases(self) -> None:
+        """Test that acompletion_iter handles various logprobs edge cases gracefully."""
+        model = LiteLLMModel(name=CommonLLMNames.OPENAI_TEST.value)
+        messages = [Message(content="Say hello", role="user")]
```
Comment (jamesbraza) [inline packages/lmi/tests/test_llms.py]:
```suggestion
        messages = [Message(content="Say hello")]
```

`"user"` is the default

File: packages/lmi/tests/test_llms.py [hunk 5]
```diff
@@ -391,6 +391,90 @@ def test_pickling(self, tmp_path: pathlib.Path) -> None:
...
+            # Mock completion with logprobs but no content
+            mock_logprobs_no_content = Mock()
+            mock_logprobs_no_content.content = None
+            mock_completion_no_content = _build_mock_completion(
+                logprobs=mock_logprobs_no_content, delta_content=" world"
+            )
+
+            # Mock completion with empty content list
+            mock_logprobs_empty = Mock()
+            mock_logprobs_empty.content = []
```
Comment (jamesbraza) [inline packages/lmi/tests/test_llms.py]:
```suggestion
            mock_logprobs_empty = Mock(content=[])
```

Let's have these be one liners, easier to read imo

Comment (alexandonian) [reply packages/lmi/tests/test_llms.py]: Absolutely

File: packages/lmi/tests/test_llms.py [hunk 6]
```diff
@@ -391,6 +391,90 @@ def test_pickling(self, tmp_path: pathlib.Path) -> None:
...
+            usage: Any = None,
+        ) -> Mock:
+            mock_completion = Mock()
+            mock_completion.model = model
+            mock_completion.choices = [Mock(logprobs=logprobs)]
+            mock_completion.choices[0].delta = Mock(
+                content=delta_content,
+                reasoning_content=delta_reasoning_content,
+                role=delta_role,
+            )
```
Comment (jamesbraza) [inline packages/lmi/tests/test_llms.py]:
```suggestion
            mock_completion = Mock(
                model=model,
                choices=[Mock(
                    logprobs=logprobs,
                    delta=Mock(
                        content=delta_content,
                        reasoning_content=delta_reasoning_content,
                        role=delta_role,
                )]
            )
```

Can we compress this into one line

File: packages/lmi/tests/test_llms.py [hunk 7]
```diff
@@ -391,6 +391,90 @@ def test_pickling(self, tmp_path: pathlib.Path) -> None:
...
+            # Mock completion with valid logprobs
+            mock_logprobs_valid = Mock()
+            mock_logprobs_valid.content = [Mock()]
+            mock_logprobs_valid.content[0].logprob = -0.5
+            mock_completion_valid = _build_mock_completion(logprobs=mock_logprobs_valid)
+
+            # Mock completion with usage info
+            mock_usage = Mock()
+            mock_usage.prompt_tokens = 10
+            mock_usage.completion_tokens = 5
```
Comment (jamesbraza) [inline packages/lmi/tests/test_llms.py]: Same here can you have this be a one liner

There's other instances in the test too

**Review by jamesbraza — APPROVED** [review]
LGTM, I think still a few more `Mock` constructions we can tighten up

File: packages/lmi/tests/test_llms.py (hunk 7 above)
Comment (alexandonian) [reply packages/lmi/tests/test_llms.py]: Okay, these should be fixed now. I thought these had been pushed already.
